sys-devel/polly/polly-19.0.0.9999.ebuild

```diff
@@ -0,0 +1,67 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit cmake llvm.org python-any-r1
+
+DESCRIPTION="Polyhedral optimizations for LLVM"
+HOMEPAGE="https://polly.llvm.org/"
+
+LICENSE="Apache-2.0-with-LLVM-exceptions UoI-NCSA"
+SLOT="${LLVM_MAJOR}/${LLVM_SOABI}"
+IUSE="+debug test"
+RESTRICT="!test? ( test )"
+
+DEPEND="
+	~sys-devel/llvm-${PV}:${LLVM_MAJOR}=[debug=]
+"
+RDEPEND="
+	${DEPEND}
+"
+BDEPEND="
+	test? (
+		$(python_gen_any_dep ">=dev-python/lit-${PV}[\${PYTHON_USEDEP}]")
+	)
+"
+
+LLVM_COMPONENTS=( polly cmake )
+llvm.org_set_globals
+
+python_check_deps() {
+	python_has_version ">=dev-python/lit-${PV}[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+}
+
+src_prepare() {
+	# prepend the newly built test binaries
+	if use test; then
+		local polly_test_bin="${WORKDIR}/${PN}_build/bin"
+		sed -i -E -e "s|^(llvm_config.add_tool_substitutions\(tool_patterns)|\1,\[\'${polly_test_bin}\',llvm_config.config.llvm_tools_dir\]|" test/lit.cfg || die "sed: couldn't add test bin search dir"
```

Comment:
This is really asking for an upstream patch. Could you send a PR there? I can do a quick review, if you need me to.

Reply:
@xarblu can you please take a look at this request? Since those lines came from your original work

Reply:
I think this was needed because the system `lit` uses the system LLVM config to find binary paths and the `polly-isl-test` binary isn't present there (or at least it's not the one we want to test). That said tests were broken anyways (for me at least) because out-of-tree polly builds want to load `LLVMPolly.so` with `-load` which  breaks if `libLLVM.so` already has Polly linked.

Reply:
> I think this was needed because the system `lit` uses the system LLVM config to find binary paths and the `polly-isl-test` binary isn't present there (or at least it's not the one we want to test). That said tests were broken anyways (for me at least) because out-of-tree polly builds want to load `LLVMPolly.so` with `-load` which breaks if `libLLVM.so` already has Polly linked.

But in this case Polly is not linked to `libLLVVM.so`, will that change something?

Reply:
Out-of-tree builds would still need that extra path to find `polly-isl-test`. I'm not sure how I would approach that upstream though because you probably still want the default path for in-tree builds.